packages/backend/src/handlers/actions/guild/saveGuildHandler.ts

```diff
@@ -23,17 +31,37 @@ export const saveGuildHandler = async (
   const { input, session_variables: sessionVariables } = req.body;
   const playerId = sessionVariables['x-hasura-user-id'];
 
+  const { guildInformation } = input;
+
+  let updatedGuild: GuildFragment;
   try {
-    const { guildInformation } = input;
-    await saveGuild(playerId, guildInformation as GuildInfoInput);
-    res.json({ success: true });
+    updatedGuild = await saveGuild(
+      playerId,
+      guildInformation as GuildInfoInput,
+    );
   } catch (error) {
     console.error(error);
     res.json({
       success: false,
       error: (error as Error).message,
     });
+    return;
   }
+
+  const isNew = updatedGuild.status === GuildStatus_Enum.Pending;
+
+  if (isNew) {
+    try {
+      await sendDiscordNotification(guildInformation);
```

Comment:
I'd name this function something more specific to what it is doing:

```suggestion
      await requestChampsApproval(guildInformation);
```

That or I'd not have the method know that it's dealing with a guild:

```suggestion
      const queryParam = encodeURIComponent(`guildname;$eq;${guildInfo.guildname}`)
      const link = (
        CONFIG.hasuraAdminURL
        + '/data/schema/public/tables/guild/browse'
        + `?filter=${queryParam}`
      );
      const message = (
        `A new guild signed up named: "${guildInfo.name}"!`
        + `\n\nSomeone with access to the Hasura instance can approve them here: ${link}`
      );
      await sendDiscordNotification(message);
```